hw/ip/prim/doc/prim_keccak.md

```diff
@@ -0,0 +1,79 @@
+----
+title: "Primitive Component: Keccak permutation"
+----
+
+# Overview
+
+`prim_keccak` is a single round implementation of the permutation stage in [SHA3 algorithm][fibs-pub-202].
+Keccak primitive module assumes the Nr is less than or equal to 12 + 2L.
+It supports all combinations of the data width described in the [spec][fibs-pub-202].
+The implementation is not hardened against side-channel or fault injection attacks.
+It implements the Keccak_p function.
+
+[fibs-pub-202]: https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.202.pdf
+
+## Parameters
+
+Name  | Type | Description
+------|------|--------------
+Width | int  | state width. can be 25, 50, 100, 200, 400, 800, or 1600
+
+### Derived Parameters
+
+Name  | Type | Description
+------|------|--------------
+W     | int  | number of slices in state. `Width/25`
+L     | int  | log2 of `W`
+Round | int  | maximum allowed round value, assumed max of `12 + 2L`
```

Comment:
I think this got renamed. And since it is derived, shouldn't you remove "assumed max of"
and replace with "equal to"?